PortalFetch/crawler_sel.py

```diff
@@ -0,0 +1,261 @@
+    submitD(driver)
+
+
+def submitD(driver):
+    """Locate submit button and submit."""
+    ipts = driver.find_elements_by_tag_name("input")  # list of Web Element
+    submit = ""  # webElement
+    for ipt in ipts:
+        if "submit" == ipt.get_attribute("type").lower() and \
+                "Section Search".lower() == ipt.get_attribute("value").lower():
+            submit = ipt
+            break
+
+    if not submit:
+        print("No submit button found! ! ")
+        sys.exit(-1)  # Program exit
+
+    #Click the submit button
+    submit.click()
+    # Wait for a while, wait for the page to open and make sure it loads
+    time.sleep(3)
+
+# Save the results of the course
+def saveResult(driver):
+    """Save result to a html."""
+    waitUtilPageLoaded(driver)
+    # Scroll down a page to show the table (no practical meaning, just look at the effect)
+    actions = ActionChains(driver)
+    actions.send_keys(Keys.PAGE_DOWN).perform()
+    # After loading is complete, the entire form page
+    html = driver.page_source
+    # print(html)
+    time.sleep(5)  # Wait for a while, let people take a look, in fact, it does not make sense to the programmer
+    return html
+
+
+#
+# Wait until the page is loaded! This only applies to child pages, not to the main page, because there is no banner_copyright information in the main page
+def waitUtilPageLoaded(driver):
+    """Wait until page loaded."""
+    while True:
+        try:
+            driver.find_element_by_class_name("banner_copyright")
+            break
+        except:
+            print("retry after 1 second...")
+            time.sleep(1)
+            pass
+
+
+# input: text is html of the courses, ulist is an empty list []
+# output : ulist will include all the courses list in text
+def get_contents(ulist, text):
+    """Get result contents from html."""
+    soup = BeautifulSoup(text, 'lxml')
+    trs = soup.find_all('tr')
+    for tr in trs:
+        ui = []
+        for td in tr:
+            ui.append(td.string)
+        ulist.append(ui)
+
+
+
+# input: urlist is the courses list we get from get_contents
+# output : save courses in a file "DeAnza2019fall.csv"
+def save_contents(urlist):
+    """Save course result to a .csv file."""
+    with open("DeAnza2019fall_sel.csv", 'w') as f:
+        writer = csv.writer(f)
+        writer.writerow(['DeAnza 2019 fall'])
+
+        for i in range(8,len(urlist)-1):
+            if len(urlist[i]) >=35:
+                writer.writerow([urlist[i][1], urlist[i][3], urlist[i][5],urlist[i][7], urlist[i][9], urlist[i][11],
+                            urlist[i][13], urlist[i][15], urlist[i][17],urlist[i][19], urlist[i][21], urlist[i][23],
+                            urlist[i][25], urlist[i][27], urlist[i][29],urlist[i][31], urlist[i][33], urlist[i][35]]
+                            )
+            else:
+                writer.writerow(urlist[i])
+    print("Finished!")
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
maybe use package `argparse` to read some of parameter

Reply:
Do you mean parameters like quarter, filename?